### PR TITLE
Make GemTable properties required

### PR DESCRIPTION
**Pull request by maxhutch**
# Citrine Python PR

## Description 
They were listed as optional, and its unclear why.  This led
to at least one bug.  This makes them required and updates
codepaths to avoid circumstnaces where any part of them
where missing.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [x] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in setup.py

### CODE REVIEW

**Comment by maxhutch** [comment]
@kroenlein I ran the e2es against this branch and it passed.

**Review by kroenlein — COMMENTED** [review]
Seems coherent, but is this philosophically consistent with

https://github.com/CitrineInformatics/citrine-python/blob/730f0aab65f964c7d174dcf192ae4cd01696b8fb/src/citrine/resources/gemtables.py#L344

**Comment by maxhutch** [comment]
> Seems coherent, but is this philosophically consistent with
> 
> https://github.com/CitrineInformatics/citrine-python/blob/730f0aab65f964c7d174dcf192ae4cd01696b8fb/src/citrine/resources/gemtables.py#L344

What do you mean?  What could be inconsistent?

File: src/citrine/resources/gemtables.py [hunk 1]
```diff
@@ -32,12 +32,12 @@ class GemTable(Resource['Table']):
     _response_key = 'table'
     _resource_type = ResourceTypeEnum.TABLE
 
-    uid = properties.Optional(properties.UUID(), 'id')
+    uid = properties.UUID('id')
     """:Optional[UUID]: unique Citrine id of this GEM Table"""
```
Comment (kroenlein) [inline src/citrine/resources/gemtables.py:36]:
```suggestion
    """:UUID: unique Citrine id of this GEM Table"""
```

File: src/citrine/resources/gemtables.py [hunk 2]
```diff
@@ -32,12 +32,12 @@ class GemTable(Resource['Table']):
...
-    version = properties.Optional(properties.Integer, 'version')
+    version = properties.Integer('version')
     """:Optional[int]: Version number of the GEM Table.
```
Comment (kroenlein) [inline src/citrine/resources/gemtables.py:38]:
```suggestion
    """:int: Version number of the GEM Table.
```

File: src/citrine/resources/gemtables.py [hunk 3]
```diff
@@ -32,12 +32,12 @@ class GemTable(Resource['Table']):
...
     The first table built from a given config is version 1."""
-    download_url = properties.Optional(properties.String, 'signed_download_url')
+    download_url = properties.String('signed_download_url')
     """:Optional[str]: Url pointing to the location of the GEM Table's contents.
```
Comment (kroenlein) [inline src/citrine/resources/gemtables.py:41]:
```suggestion
    """:str: URL pointing to the location of the GEM Table's contents.
```

**Review by kroenlein — CHANGES_REQUESTED** [review]
Arguments are no longer optional

**Review by kroenlein — APPROVED** [review]
LGTM

**Pull request merged**